libtsuba/src/tsuba.cpp

```diff
@@ -146,21 +146,23 @@ ParseManifestName(
 
 }  // namespace
 
-// TODO (scober, KAT-1870): write a test for this
 katana::Result<std::pair<uint64_t, std::vector<tsuba::RDGView>>>
-tsuba::ListAvailableViews(
+tsuba::ListViewsOfVersion(
     const std::string& rdg_dir, std::optional<uint64_t> version) {
   auto rdg_uri = KATANA_CHECKED(katana::Uri::Make(rdg_dir));
   std::vector<std::string> files = KATANA_CHECKED(FileList(rdg_uri.string()));
 
   std::vector<tsuba::RDGView> views_found;
   uint64_t latest_version = 0;
+  bool any_version_found = false;
   for (const std::string& file : files) {
     std::string view_type;
     std::vector<std::string> view_args;
     uint64_t view_version{};
 
-    if (!ParseManifestName(file, &view_type, &view_args, &view_version)) {
+    if (ParseManifestName(file, &view_type, &view_args, &view_version)) {
+      any_version_found = true;
```

Comment:
This control flow was hard for me to parse the first time, and I think it's wrong. We should also return an error if you ask for a particular version and it doesn't exist. Could we instead flip this back and have: 
```c++
if (!ParseManifestName(file, &view_type, &view_args, &view_version)) {
  continue;
}

// if a specific version was requested, only return views for that version
if (version && view_version != version.value()) {
  continue;
}
any_version_found = true:
```

Reply:
Why don't we revert back to the pre #575 version or fix the empty-graph corner case in katana-enterprise?

Reply:
My goal was to distinguish between non-RDG directories and RDG directories where we didn't find the right view. But it doesn't seem like we care about that distinction.

Reply:
It's different than that right? That distinction is still important, this is about distinguishing between a non-rdg directory and one were we didn't find the right version. I think if a specific version is asked for it's that last distinction that does not matter.